server/lorax_server/utils/graph.py

```diff
@@ -603,7 +605,13 @@ def forward(
 
         key = (batch_size, max_rank)
         graph = self.cache.get(key)
-        if graph is None or not graph.input_state.traced_adapter_layer_names.issuperset(adapter_data.layer_names()):
+        if (
+            graph is None
+            or not graph.input_state.traced_adapter_layer_names.issuperset(adapter_data.layer_names())
+            # This is the case where COMPILE_BATCH_SIZE < batch_size <= MAX_BATCH_SIZE so
+            # we just retrace the graph for that new size
+            or batch_size > self.batch_size
```

Comment:
https://github.com/predibase/lorax/blob/12e530a0be722871be1cfb3e791f9aed48f66dca/server/lorax_server/utils/graph.py#L478-L498

This ensures that we stay within the range